src/rpc/server/scala/serv/src/main/scala/com/ctc/omega_edit/grpc/EditorService.scala

```diff
@@ -459,6 +459,29 @@ class EditorService(implicit val system: ActorSystem) extends Editor {
             SegmentResponse.of(in.sessionId, offset, ByteString.copyFrom(data))
           )
       }
+
+  def stopServer(kind: ServerControlKind) = {
+    system.terminate()
+    Future.successful(ServerControlResponse(kind, 0))
```

Comment:
You should map the terminate call into the ServerControlResponse.  As is, you have a race condition here.

Reply:
@jw3 Believe the updated code should eliminate this correct? If successfully terminated set 0 and if failure then set 1.